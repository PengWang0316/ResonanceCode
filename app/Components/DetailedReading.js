import React from "react";
import PropTypes from "prop-types";
import LineBigram from "./LineBigram";
// import QueryString from "query-string";

const DetailedReading = (props)=>{

  // componentWillMount(){
  //   let queryInfo=QueryString.parse(this.props.location.search);
  //   console.log("DetailedReading page queryInfo:",queryInfo);
  // }

    const renderArray=[];
    /* Exchanging the order for line 46 and line 25 */
    const newArray = [props.imageInfos[0], props.imageInfos[2], props.imageInfos[1]]
    newArray.map((element,index)=>{
      renderArray.push(<LineBigram key={index} title={element.title} data={element} />);
    });

    return(
      <div className="detailContainer">
        {renderArray}
      </div>
    );

};
DetailedReading.proptypes={
  imageInfos: PropTypes.array.isRequired
};
export default DetailedReading;
