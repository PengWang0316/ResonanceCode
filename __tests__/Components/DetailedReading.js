import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import DetailedReading from '../../app/Components/DetailedReading';

jest.mock('../../app/Components/LineBigram', () => 'LineBigram');
jest.mock('../../app/Components/AssociatedHexagrams', () => 'AssociatedHexagrams');
jest.mock('../../app/Components/BigramClockSmall', () => 'BigramClockSmall');
jest.mock('../../app/Components/BigramBlockSmall', () => 'BigramBlockSmall');

describe('DetailedReading test', () => {
  const defaultProps = {
    imageInfos: [{ _id: '1', title: 'title1' }, { _id: '2', title: 'title2' }, { _id: '3', title: 'title3' }],
    hexagram: {
      resonance_bigram_question: 'resonance_bigram_question', wave_bigram_question: 'wave_bigram_question', paticle_bigram_question: 'paticle_bigram_question', analysis: 'analysis', question: 'question', overview: 'overview', number: 1
    },
    handleHexagramClick: jest.fn()
  };

  const getShallowComponent = () => shallow(<DetailedReading {...defaultProps} />);

  test('DetailedReading snapshot', () => expect(renderer.create(<DetailedReading {...defaultProps} />).toJSON()).toMatchSnapshot());

  test('handleHexagramClick', () => {
    const component = getShallowComponent();
    component.find({ id: 1 }).simulate('click');
    expect(defaultProps.handleHexagramClick).toHaveBeenCalledTimes(1);
  });
});
