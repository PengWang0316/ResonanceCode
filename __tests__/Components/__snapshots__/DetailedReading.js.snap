// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailedReading test DetailedReading snapshot 1`] = `
<div
  className="mt-3"
>
  <div
    className="detailButton"
    id="img_arr"
    onClick={[MockFunction]}
    role="button"
    tabIndex="-1"
  >
    <i
      className="fas fa-book mr-2"
    />
    Show Detailed Hexagram
  </div>
  <div
    className="d-flex"
  >
    <BigramClockSmall
      lineText="Line 2/5"
      position={undefined}
    />
    <BigramClockSmall
      lineText="Line 1/4"
      position={undefined}
    />
    <BigramClockSmall
      lineText="Line 3/6"
      position={undefined}
    />
  </div>
  <div
    className="d-flex mt-1 mb-4"
  >
    <BigramBlockSmall
      lineText="Line 1/3"
      position={undefined}
    />
    <BigramBlockSmall
      lineText="Line 4/6"
      position={undefined}
    />
  </div>
  <div
    className="mt-2 font-weight-bold"
  >
    Overview:
  </div>
  <div>
    overview
  </div>
  <div
    className="mt-2 font-weight-bold"
  >
    Analysis:
  </div>
  <div>
    analysis
  </div>
  <div
    className="mt-2 font-weight-bold"
  >
    Question:
  </div>
  <div>
    question
  </div>
</div>
`;
